src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/AbstractSyntaxFactsService.cs

```diff
@@ -539,13 +546,13 @@ public bool IsOnHeader<THoleSyntax>(int position, SyntaxNode ownerOfHeader, Synt
             return tokenToRightOrIn.GetPreviousToken().GetAncestor<TNode>();
         }
 
-        protected int GetStartOfNodeExcludingAttributes(SyntaxNode node)
+        protected int GetStartOfNodeExcludingAttributes(SyntaxNode root, SyntaxNode node)
         {
             var attributeList = GetAttributeLists(node);
             if (attributeList.Any())
             {
                 var endOfAttributeLists = attributeList.Last().Span.End;
-                var afterAttributesToken = node.FindTokenOnRightOfPosition(endOfAttributeLists);
+                var afterAttributesToken = root.FindTokenOnRightOfPosition(endOfAttributeLists);
```

Comment:
why can't hte root be determined from the node passed in?

Reply:
IIRC it was the pattern of other API calls and the root was always available at the callsite. As such passing it instead of determining it every time (IIRC not a constant op (even though probably cheap O(depth of syntaxtree) enough)) was deemed better.

Also, it was the last day of my internship so I think larger refactoring of the above-mentioned pattern wasn't considered worth it (not saying it's a great justification for cutting corners, ofc).

Reply:
Also the reason why I haven't noticed this message, sorry :)